log.go

```diff
@@ -56,8 +54,13 @@ type ZapEventLogger struct {
 // that are provided to the logger via SetFieldsOnAllLoggers and these
 // only last for the life time of this particular ZapEventLogger instance.
 // Note: the fields will be passed to any children loggers of this logger.
-func (zel *ZapEventLogger) SetFieldsOnLogger(name string, args ...interface{}) {
-	zel.SugaredLogger = *zel.With(args...)
+func (zel *ZapEventLogger) SetFieldsOnLogger(args ...interface{}) {
+	loggerMutex.Lock()
+	defer loggerMutex.Unlock()
```

Comment:
This lock protects the `loggers` map and that's _all_ it protects. There's absolutely no reason `loggers[zel.system].SugaredLogger = ...` would be any safer than `zel.SugaredLogger = ...`.